node/actors/bft/src/lib.rs

```diff
@@ -96,34 +90,27 @@ impl Config {
             // This is the infinite loop where the consensus actually runs. The validator waits for either
             // a message from the network or for a timeout, and processes each accordingly.
             loop {
-                let deadline = *deadline_recv.borrow();
-                let input = pipe.recv.recv(&ctx.with_deadline(deadline)).await.ok();
+                let input = pipe.recv.recv(ctx).await;
 
                 // We check if the context is active before processing the input. If the context is not active,
                 // we stop.
                 if !ctx.is_active() {
                     return Ok(());
                 }
 
-                // Check if an asynchronous error was received.
+                // Check if an asynchronous error was returned.
                 if let Ok(res) = err_recv.try_recv() {
                     return Err(res.err().unwrap());
                 }
 
-                let Some(InputMessage::Network(req)) = input else {
-                    let res_recv = replica_send.enqueue(None).await;
-                    // Wait for result before proceeding, allowing timeout deadline value to get updated.
-                    handle_result(ctx, res_recv, &err_send).await;
-                    continue;
-                };
-
+                let InputMessage::Network(req) = input.unwrap();
                 let res_recv;
                 match &req.msg.msg {
                     ConsensusMsg::ReplicaPrepare(_) | ConsensusMsg::ReplicaCommit(_) => {
-                        res_recv = leader_send.enqueue(req.msg).await;
+                        res_recv = leader_send.send(req.msg).await;
                     }
                     ConsensusMsg::LeaderPrepare(_) | ConsensusMsg::LeaderCommit(_) => {
-                        res_recv = replica_send.enqueue(Some(req.msg)).await;
+                        res_recv = replica_send.send(req.msg).await;
```

Comment:
can you pass the response channel together with the message to the replica/leader, so that you don't have to have handle_result at all?

Reply:
Yes, especially if we don't need to be able to return errors (https://github.com/matter-labs/era-consensus/pull/56#discussion_r1462249244).

Reply:
Fixed, changes here: https://github.com/matter-labs/era-consensus/pull/56/commits/2209d5dc32e7c6ce4195c19101faf4290b68535c